server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedNumericIndexFieldData.java

```diff
@@ -113,95 +101,29 @@ public final Index index() {
         return index;
     }
 
-    /**
-     * Returns the {@link SortField} to used for sorting.
-     * Values are casted to the provided <code>targetNumericType</code> type if it doesn't
-     * match the field's <code>numericType</code>.
-     */
-    public SortField sortField(NumericType targetNumericType, Object missingValue, MultiValueMode sortMode,
-                               Nested nested, boolean reverse) {
-        final XFieldComparatorSource source = comparatorSource(targetNumericType, missingValue, sortMode, nested);
-
-        /**
-         * Check if we can use a simple {@link SortedNumericSortField} compatible with index sorting and
-         * returns a custom sort field otherwise.
-         */
-        if (nested != null
-                || (sortMode != MultiValueMode.MAX && sortMode != MultiValueMode.MIN)
-                || numericType == NumericType.HALF_FLOAT
-                || targetNumericType != numericType) {
-            return new SortField(fieldName, source, reverse);
-        }
-
-        final SortField sortField;
-        final SortedNumericSelector.Type selectorType = sortMode == MultiValueMode.MAX ?
-            SortedNumericSelector.Type.MAX : SortedNumericSelector.Type.MIN;
-        switch (numericType) {
-            case FLOAT:
-                sortField = new SortedNumericSortField(fieldName, SortField.Type.FLOAT, reverse, selectorType);
-                break;
-
-            case DOUBLE:
-                sortField = new SortedNumericSortField(fieldName, SortField.Type.DOUBLE, reverse, selectorType);
-                break;
-
-            default:
-                assert !numericType.isFloatingPoint();
-                sortField = new SortedNumericSortField(fieldName, SortField.Type.LONG, reverse, selectorType);
-                break;
-        }
-        sortField.setMissingValue(source.missingObject(missingValue, reverse));
-        return sortField;
-    }
-
     @Override
-    public SortField sortField(Object missingValue, MultiValueMode sortMode, Nested nested, boolean reverse) {
-        return sortField(numericType, missingValue, sortMode, nested, reverse);
-    }
-
-    /**
-     * Builds a {@linkplain BucketedSort} for the {@code targetNumericType},
-     * casting the values if their native type doesn't match.
-     */
-    public BucketedSort newBucketedSort(NumericType targetNumericType, BigArrays bigArrays, @Nullable Object missingValue,
-            MultiValueMode sortMode, Nested nested, SortOrder sortOrder, DocValueFormat format,
-            int bucketSize, BucketedSort.ExtraData extra) {
-        return comparatorSource(targetNumericType, missingValue, sortMode, nested)
-                .newBucketedSort(bigArrays, sortOrder, format, bucketSize, extra);
+    protected boolean sortRequiresCustomComparator() {
+        return numericType == NumericType.HALF_FLOAT;
```

Comment:
Maybe leave a comment in here about why half_floats need a custom comparator?  I'm assuming their encoding makes them non-ordered or something?

Reply:
I believe that is so. I'll dig and leave a comment.